### PR TITLE
I added cuphead (made by me)

### PR DESCRIPTION
**Pull request by Alekemon**
Feel free to add them!

### CODE REVIEW

File: Source/SmashBros/Game/Characters/Cuphead.h [hunk 1]
```diff
@@ -0,0 +1,100 @@
+
+#include "../../Player.h"
+
+#pragma once
+
+namespace SmashBros
+{
+	class Fox : public Player
+	{
```
Comment (lufinkey) [inline Source/SmashBros/Game/Characters/Cuphead.h:9]: looks like you may have forgotten to rename this spot ^

**Comment by lufinkey** [comment]
Damn man this is really impressive. Do you mind listing out the movesets and all that, and maybe including some gifs of them in action? I'll try it out a bit later but this is awesome

File: Source/SmashBros/Game/Characters/Kirby.cpp [hunk 2]
```diff
@@ -0,0 +1,450 @@
...
+				{
+					AttackTemplates::normalAirDownA(this, 10,2.052f);
+				}
+			}
+		}
+	}
+
+	void Kirby::attackB()
+	{
+		//
```
Comment (lufinkey) [inline Source/SmashBros/Game/Characters/Kirby.cpp:359]: probably wanna add a B attack before merging

Comment (Alekemon) [reply Source/SmashBros/Game/Characters/Kirby.cpp:359]: Uh right

File: Source/SmashBros/Game/Characters/Cuphead.h (hunk 1 above)
Comment (Alekemon) [reply Source/SmashBros/Game/Characters/Cuphead.h:9]: I took Fox as a base, so thanks!

**Comment by Alekemon** [comment]
Removed Kirby and revised for missing code. Thanks!

**Comment by lufinkey** [comment]
I'm still getting a couple compile errors. I can try to set this up as a codelite project for you later so you can work on it on Linux. I'd definitely want you to have a real dev environment if you're trying to add stuff to it

**Comment by lufinkey** [comment]
Hey if you're adding more than one character, you should make it as a separate branch / PR, so we can vet them individually

**Comment by Alekemon** [comment]
for me compiling it with cuphead worked

**Comment by Alekemon** [comment]
I added rayman because of global.cpp and loader.h having shared character values

**Comment by Alekemon** [comment]
> I'm still getting a couple compile errors. I can try to set this up as a codelite project for you later so you can work on it on Linux. I'd definitely want you to have a real dev environment if you're trying to add stuff to it

yes pls. that way I could add classic mode easily, as I'd like to do

**Comment by lufinkey** [comment]
Do you mind making Rayman as a separate PR and branch though? It's just a lot easier for me to deal with them individually

**Comment by Alekemon** [comment]
Ok, revise Cuphead on this one. I'll create Rayman on a second one

**Comment by lufinkey** [comment]
Thanks!

**Comment by Alekemon** [comment]
Ur welcome

**Comment by Alekemon** [comment]
done

**Comment by Alekemon** [comment]
How are you going?

**Comment by lufinkey** [comment]
I'm at work right now. I'll get the codelite project made when I get off

**Comment by Alekemon** [comment]
Uh right. Thanks for letting me know!

**Comment by lufinkey** [comment]
Just added a codelite project that you can use for linux builds

**Comment by lufinkey** [comment]
Also sorry for the messy code heh I wrote this like 7 years ago

**Comment by lufinkey** [comment]
Also it looks like you may be missing the grab animation

**Comment by Alekemon** [comment]
Uh right. I'll check the grab animation

**Comment by Alekemon** [comment]
There's a new pull request

**Pull request closed without merging**